deploy/eck-elasticsearch/values.yaml

```diff
@@ -85,11 +137,208 @@ nodeSets:
   podTemplate:
+    # The following spec is exactly the Kubernetes Core V1 PodTemplateSpec. Any fields within the PodTemplateSpec
+    # are supported within the 'spec' field below.  Please see below documentation for the exhaustive list of fields.
+    #
+    # https://v1-24.docs.kubernetes.io/docs/reference/generated/kubernetes-api/v1.24/#podtemplatespec-v1-core
+    #
+    # Only the commonly overridden/used fields will be noted below.
+    # 
     spec:
+
+    # If specified, the pod's scheduling constraints
+    # https://www.elastic.co/guide/en/cloud-on-k8s/current/k8s-advanced-node-scheduling.html
+    # affinity:
+    #   nodeAffinity:
+    #     requiredDuringSchedulingIgnoredDuringExecution:
+    #       nodeSelectorTerms:
+    #       - matchExpressions:
+    #         - key: topology.kubernetes.io/zone
+    #           operator: In
+    #           values:
+    #           - antarctica-east1
+    #           - antarctica-west1
+    #     preferredDuringSchedulingIgnoredDuringExecution:
+    #     - weight: 1
+    #       preference:
+    #         matchExpressions:
+    #         - key: another-node-label-key
+    #           operator: In
+    #           values:
+    #           - another-node-label-value
+    #   podAffinity:
+    #     requiredDuringSchedulingIgnoredDuringExecution:
+    #     - labelSelector:
+    #         matchExpressions:
+    #         - key: security
+    #           operator: In
+    #           values:
+    #           - S1
+    #       topologyKey: topology.kubernetes.io/zone
+    #   podAntiAffinity:
+    #     preferredDuringSchedulingIgnoredDuringExecution:
+    #     - weight: 100
+    #       podAffinityTerm:
+    #         labelSelector:
+    #           matchLabels:
+    #             elasticsearch.k8s.elastic.co/cluster-name: quickstart
+    #         topologyKey: kubernetes.io/hostname
+
+    # Containers array.  Should only be used to customize the 'elasticsearch' container using the following fields.
       containers:
       - name: elasticsearch
+
+        # List of environment variables to set in the 'elasticsearch' container.
+        # env:
+        # name/value environment variable pair to set within the 'elasticsearch' container.
+        # - name: "my-env-var"
+        #   value: "my-value"
+        # Set a named environment variable's value from a configuration source.
+        # - name: "my-env-from"
+        #   valueFrom:
+        #   Set the value from a configmap.
+        #   configMapKeyRef:
+        #     name: specialEnvConfig
+        #   Set the value from the container's field.
+        #   fieldRef:
+        #     fieldPath: spec.nodeName
+        #   Set the value from the container's resource requests.
+        #   resourceFieldRef:
+        #     containerName: "elasticsearch"
+        #     resource: requests.cpu
+        #   Set the value from a secret.
+        #   secretKeyRef:
+        #     name: specialEnvSecretConfig
+        #     key: "my-key"
```

Comment:
Same here with the env vars (above and below my comment)  I would use a shorter example and maybe a link to the K8s docs.